staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/grpc_service.go

```diff
@@ -108,27 +98,33 @@ func parseEndpoint(endpoint string) (string, error) {
 
 // Check the KMS provider API version.
 // Only matching kmsapiVersion is supported now.
-func checkAPIVersion(kmsClient kmsapi.KeyManagementServiceClient) error {
-	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+func (g *gRPCService) checkAPIVersion() error {
+	ctx, cancel := context.WithTimeout(context.Background(), g.callTimeout)
 	defer cancel()
 
 	request := &kmsapi.VersionRequest{Version: kmsapiVersion}
-	response, err := kmsClient.Version(ctx, request)
+	response, err := g.kmsClient.Version(ctx, request)
 	if err != nil {
 		return fmt.Errorf("failed get version from remote KMS provider: %v", err)
 	}
 	if response.Version != kmsapiVersion {
-		return fmt.Errorf("KMS provider api version %s is not supported, only %s is supported now",
-			response.Version, kmsapiVersion)
+		return fmt.Errorf(versionErrorf, response.Version, kmsapiVersion)
 	}
 
 	glog.V(4).Infof("KMS provider %s initialized, version: %s", response.RuntimeName, response.RuntimeVersion)
+	g.versionChecked = true
```

Comment:
1. It's weird to set this here but make every caller check it?
2. Shouldn't this be locked or thread safe in some other way?